src/app/signup/page.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+
+export default function SignupPage() {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+
+    const router = useRouter();
+
+    async function handleSubmit(e: React.FormEvent) {
+        e.preventDefault();
+
+        const res = await fetch("/api/auth/signup", {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ email, password }),
+        });
+
+        const data = await res.json();
+        alert(data.message || data.error);
+
+        if (res.ok) {
+            // Redirect to login page
+            // window.location.href = "/login";
+            router.push("/login");
+
+        } else {
+            alert(data.error);
+        }
```

Comment:
Users see two alert popups when signup fails. The first alert shows the message, then immediately another alert shows the same error.

```suggestion
const data = await res.json();

if (res.ok) {
    alert(data.message || "Account created successfully!");
    router.push("/login");
} else {
    alert(data.error || "Signup failed. Please try again.");
}
```